spec/unit/tasks/message_queue_spec.rb

```diff
@@ -26,6 +26,52 @@
     end
   end
 
+  describe "message_queue:insert_data_into_govuk" do
+    it "uses GovukMessageQueueConsumer::Consumer" do
+      processor = double
+      expect(GovukIndex::PublishingEventProcessor).to receive(:new).and_return(processor)
+
+      consumer = double("consumer")
+      expect(consumer).to receive(:run).and_return(true)
+
+      logger = Logging.logger[GovukIndex::PublishingEventProcessor]
+
+      expect(GovukMessageQueueConsumer::Consumer).to receive(:new)
+        .with(
+          queue_name: "search_api_govuk_index",
+          processor:,
+          logger:,
+          worker_threads: 10,
+          prefetch: 10,
+        ).and_return(consumer)
+
+      Rake::Task["message_queue:insert_data_into_govuk"].invoke
+    end
+  end
+
+  describe "message_queue:bulk_insert_data_into_govuk" do
+    it "uses GovukMessageQueueConsumer::Consumer" do
+      processor = double
+      expect(GovukIndex::PublishingEventProcessor).to receive(:new).and_return(processor)
+
+      consumer = double("consumer")
+      expect(consumer).to receive(:run).and_return(true)
+
+      logger = Logging.logger[GovukIndex::PublishingEventProcessor]
+
+      expect(GovukMessageQueueConsumer::Consumer).to receive(:new)
+        .with(
+          queue_name: "search_api_bulk_reindex",
+          processor:,
+          logger:,
+          worker_threads: 10,
+          prefetch: 10,
+        ).and_return(consumer)
+
+      Rake::Task["message_queue:bulk_insert_data_into_govuk"].invoke
+    end
+  end
```

Comment:
Nice one adding these specs.

We might want to add a `before { Rake::Task[task_name].reenable }` before them as they currently leave a bit of a trap if anyone was to add any specs for this task as it'd not execute a second time without it.